### PR TITLE
Update uuid dependency to version 4.1.0

### LINKED ISSUE #364: Incompatible with UUID v4
Because no versions of intercom_flutter match >8.0.1 <9.0.0 and intercom_flutter 8.0.1 depends on intercom_flutter_web ^1.0.0, intercom_flutter ^8.0.1 requires intercom_flutter_web ^1.0.0.
Because intercom_flutter_web 1.0.0 depends on uuid ^3.0.7 and no versions of intercom_flutter_web match >1.0.0 <2.0.0, intercom_flutter_web ^1.0.0 requires uuid ^3.0.7.
Thus, intercom_flutter ^8.0.1 requires uuid ^3.0.7.
So, because shift_flow depends on both intercom_flutter ^8.0.1 and uuid ^4.1.0, version solving failed.

You can try the following suggestion to make the pubspec resolve:
* Consider downgrading your constraint on uuid: flutter pub add uuid:^3.0.7

### PR DESCRIPTION
**Pull request by vlazdra**
Resolves https://github.com/v3rm0n/intercom_flutter/issues/364

### CODE REVIEW

**Comment by Zazo032** [comment]
Thanks for the quick fix! Can you also bump the `version` in `pubspec.yaml` and add a new entry to the `CHANGELOG.md`?

**Comment by vlazdra** [comment]
Hey @Zazo032 👋 
I've updated the `version` in `pubspec.yaml` and the `CHANGELOG.md`.

But not sure if it's possible to update the `intercom_flutter` package dependency for the library until it's published to pub.dev?
This can be updated latter on if you want once this is published: https://github.com/v3rm0n/intercom_flutter/blob/master/intercom_flutter/pubspec.yaml#L13

**Comment by Zazo032** [comment]
Thanks! We can merge and publish the web package first, then update it in the main one

**Review by Zazo032 — APPROVED** [review]

File: intercom_flutter_web/pubspec.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ dependencies:
   flutter_web_plugins:
     sdk: flutter
   intercom_flutter_platform_interface: ^2.0.0
-  uuid: ^3.0.7 # to get the random uuid for loginUnidentifiedUser in web
+  uuid: ^4.2.1 # to get the random uuid for loginUnidentifiedUser in web
```
Comment (junaid1460) [inline intercom_flutter_web/pubspec.yaml:19]: should this be an exact match?

**Pull request merged**